app/models/tenant.rb

```diff
@@ -154,8 +154,8 @@ def set_quotas(quotas)
       end
       # Delete any quotas that were not passed in
       tenant_quotas.destroy_missing(updated_keys)
-      # unfortunatly, an extra scope is created in destroy_missing, so we need to reload the records
-      clear_association_cache
+      # unfortunately, an extra scope is created in destroy_missing, so we need to reload the records
+      reload
```

Comment:
let me verify this doesn't have the same problem with "changed" records having those changes discarded...